homeassistant/components/fritzbox/quality_scale.yaml

```diff
@@ -0,0 +1,83 @@
+rules:
+  # Bronze
+  action-setup:
+    status: exempt
+    comment: no actions/services are implemented
+  appropriate-polling: done
+  brands: done
+  common-modules: done
+  config-flow-test-coverage: done
+  config-flow:
+    status: todo
+    comment: some data descriptions are missing
+  dependency-transparency: done
+  docs-actions:
+    status: exempt
+    comment: no actions/services are implemented
+  docs-high-level-description: done
+  docs-installation-instructions: done
+  docs-removal-instructions: done
+  entity-event-setup: done
+  entity-unique-id: done
+  has-entity-name:
+    status: todo
+    comment: only set for sensors, yet
+  runtime-data: done
+  test-before-configure: done
+  test-before-setup: done
+  unique-config-entry: done
+
+  # Silver
+  action-exceptions:
+    status: todo
+    comment: add exception handling to entity cations
+  config-entry-unloading: done
+  docs-configuration-parameters:
+    status: exempt
+    comment: has no options flow
+  docs-installation-parameters: done
+  entity-unavailable: done
+  integration-owner: done
+  log-when-unavailable: done
+  parallel-updates:
+    status: todo
+    comment: not set at the moment, we use a coordinator
+  reauthentication-flow: done
+  test-coverage: done
+
+  # Gold
+  devices: done
+  diagnostics: done
+  discovery-update-info: done
+  discovery: done
+  docs-data-update: done
+  docs-examples: done
+  docs-known-limitations: done
+  docs-supported-devices: done
+  docs-supported-functions: done
+  docs-troubleshooting: done
+  docs-use-cases: done
+  dynamic-devices: done
+  entity-category: done
+  entity-device-class: done
+  entity-disabled-by-default: done
+  entity-translations: done
+  exception-translations: done
+  icon-translations: done
+  reconfiguration-flow: done
```

Comment:
While you have a reconfigure flow in fritzbox, it doesn't allow to reconfigure the username and password. Somebody needs to wait for the reauthentication flow to kick in or delete and reconfigure the config entry once again.

Reply:
the reconfiguration flow is not intended to update credentials, that's the intention for the reauth flow

Reply:
Currently the documentation mentions that it should not be used for reauthentication, but the quality scale actually says it should be included. Something needs to be changed to make it clearer and while it's being looked in what, I guess we're both right.

Reply:
The IQS rule about the [reconfiguration](https://developers.home-assistant.io/docs/core/integration-quality-scale/rules/reconfiguration-flow) says, that an integration should ideally inform the user, that a re-authentication is needed - this is done by the re-auth-flow